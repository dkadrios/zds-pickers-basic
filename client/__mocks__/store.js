import deepFreeze from 'deep-freeze'
import inputControls from './inputControls'
import mappings from './mappings'
import midi from './midi'
import shiftGroups from './shiftGroups'
import shifter from './shifter'
import user from './user'
import version from './version'

const state = { inputControls, mappings, midi, shiftGroups, shifter, user, version }

deepFreeze(state)
export default state
