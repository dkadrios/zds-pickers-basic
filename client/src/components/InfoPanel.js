import React from 'react'
import { connect } from 'react-redux'
import styles from '../styles/infoPanel.scss'
import { versionShape } from '../shapes'

export const InfoPanel = ({ version }) => {
  const { client, firmware } = version
  const formatted = value => (Number.isNaN(value) ? 'N/A' : `v${(value / 10).toFixed(1)}`)
  const year = () => new Date().getFullYear()
  const foundVersion = () => formatted(firmware)

  return (
    <footer className={styles.infoPanel}>
      <div className={styles.copyright}>
        &copy; Copyright <a href="https://zendrumstudio.com">Zendrum Studio</a>, {year()}
      </div>
      <div className={styles.versions}>
        <span className={styles.version}>Client: {formatted(client)}</span>
        <span className={styles.version}>Firmware: {foundVersion()}</span>
      </div>
    </footer>
  )
}

InfoPanel.propTypes = {
  version: versionShape.isRequired,
}

export const mapStateToProps = ({ version }) => ({ version })

export default connect(mapStateToProps)(InfoPanel)
