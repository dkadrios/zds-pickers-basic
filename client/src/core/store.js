import { applyMiddleware, combineReducers, createStore } from 'redux'
import createSagaMiddleware from 'redux-saga'
import setup from 'redux-midi-fork'
// import thunk from 'redux-thunk'
import * as reducers from '../reducers'
import sagas from '../sagas'
// import sysexInputMiddleware from '../middleware/sysexInput'
// import sysexOutputMiddleware from '../middleware/sysexOutput'
// import hardwareTestOutputMiddleware from '../middleware/hardwareTestOutput'
// import fileExport from '../middleware/fileExport'
// import fileImport from '../middleware/fileImport'
// import { watchForDeviceChange } from '../midi'
import { isDevEnv } from '../selectors'
import { sortObject } from './fp/objects'

const combinedReducers = combineReducers(sortObject(reducers))
const sagaMiddleware = createSagaMiddleware()

/* istanbul ignore next */
const createLoggerMiddleware = async () => {
  const reduxLogger = await import('redux-logger')
  return reduxLogger.createLogger({
    collapsed: true,
    diff: false,
    duration: true,
    // predicate: (getState, action) => action.type !== success(actionTypes.SESSION_KEEP_ALIVE),
    // titleFormatter: (action, time, took) => {
    //   const { options: { method } = { method: '???' }, type, url } = action
    //   const parts = ['action']
    //   parts.push(type)
    //   parts.push(`@ ${time}`)
    //   parts.push(`(${took.toFixed(2)} ms)`)
    //   if (type === actionTypes.CALL_API) {
    //     parts.push(`[${String(method).toUpperCase()}] ${url}`)
    //   }

    //   return parts.join(' ')
    // },
  })
}

const buildMiddlewares = async (isDev) => {
  const { inputMiddleware, outputMiddleware } = setup({ midiOptions: { sysex: true } })
  const result = [inputMiddleware, outputMiddleware, sagaMiddleware]
  /* istanbul ignore next */
  if (isDev) {
    result.push(await createLoggerMiddleware())
  }
  return result
}

const storeFactory = async (initialState = {}) => {
  const middlewares = await buildMiddlewares(isDevEnv())
  let applied

  /* istanbul ignore if */
  if (isDevEnv()) {
    const { composeWithDevTools } = await import('redux-devtools-extension')
    applied = composeWithDevTools(applyMiddleware(...middlewares))
  } else {
    applied = applyMiddleware(...middlewares)
  }

  const store = applied(createStore)(combinedReducers, initialState)

  sagaMiddleware.run(sagas)

  return store
}

export default storeFactory

// export default function storeFactory(initialState) {
//   const isDev = isDevEnv()
//   /* istanbul ignore if */
//   // if (!test) {
//   const { inputMiddleware, outputMiddleware } = setup({ midiOptions: { sysex: true } })
//   middlewares = [
//     ...middlewares,
//     inputMiddleware,
//     outputMiddleware,
//     sysexInputMiddleware,
//     sysexOutputMiddleware,
//     hardwareTestOutputMiddleware,
//     fileExport,
//     fileImport,
//   ]
//   // }

//   /* eslint-disable indent */
//   const store = (isDev
//     ? compose(
//         applyMiddleware(...middlewares),
//         applyMiddleware(logger),
//         window.devToolsExtension ? window.devToolsExtension() : f => f,
//       )
//     : applyMiddleware(...middlewares))(createStore)(rootReducer, initialState)
//   /* eslint-enable indent */

//   /* istanbul ignore if */
//   // if (module.hot) {
//   //   module.hot.accept('./reducers', () => {
//   //     // eslint-disable-next-line
//   //     const nextRootReducer = require('./reducers/index')
//   //     store.replaceReducer(nextRootReducer)
//   //   })
//   // }

//   /* istanbul ignore if */
//   // if (!test) {
//   /*
//      * This effectively bootstraps the application.
//      *
//      * Whenever our device is detected, either initially or due to being
//      * plugged in, watchForDeviceChange will dispatch shifterFound().
//      * This action is intercepted in /middleware/sysexInput, which then
//      * dispatches additional events to request the internal state of the
//      * device.
//      */
//   watchForDeviceChange(store)
//   // }

//   return store
// }
