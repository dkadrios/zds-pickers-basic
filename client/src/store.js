import { applyMiddleware, compose, createStore } from 'redux'
import { createLogger } from 'redux-logger'
import setup from 'redux-midi-fork'
import thunk from 'redux-thunk'
import rootReducer from './reducers'
import sysexInputMiddleware from './middleware/sysexInput'
import sysexOutputMiddleware from './middleware/sysexOutput'
import hardwareTestOutputMiddleware from './middleware/hardwareTestOutput'
import fileExport from './middleware/fileExport'
import fileImport from './middleware/fileImport'
import { watchForDeviceChange } from './midi'

const logger = createLogger({
  collapsed: true,
  diff: true,
})
let middlewares = [thunk]

export default function storeFactory(
  initialState,
  /* istanbul ignore next */ debug = __DEV__,
  /* istanbul ignore next */ test = __TEST__,
) {
  /* istanbul ignore if */
  if (!test) {
    const { inputMiddleware, outputMiddleware } = setup({ midiOptions: { sysex: true } })
    middlewares = [
      ...middlewares,
      inputMiddleware,
      outputMiddleware,
      sysexInputMiddleware,
      sysexOutputMiddleware,
      hardwareTestOutputMiddleware,
      fileExport,
      fileImport,
    ]
  }

  /* eslint-disable indent */
  const store = (debug
    ? compose(
        applyMiddleware(...middlewares),
        applyMiddleware(logger),
        window.devToolsExtension ? window.devToolsExtension() : f => f,
      )
    : applyMiddleware(...middlewares))(createStore)(rootReducer, initialState)
  /* eslint-enable indent */

  /* istanbul ignore if */
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      // eslint-disable-next-line
      const nextRootReducer = require('./reducers/index')
      store.replaceReducer(nextRootReducer)
    })
  }

  /* istanbul ignore if */
  if (!test) {
    /*
     * This effectively bootstraps the application.
     *
     * Whenever our device is detected, either initially or due to being
     * plugged in, watchForDeviceChange will dispatch shifterFound().
     * This action is intercepted in /middleware/sysexInput, which then
     * dispatches additional events to request the internal state of the
     * device.
     */
    watchForDeviceChange(store)
  }

  return store
}
