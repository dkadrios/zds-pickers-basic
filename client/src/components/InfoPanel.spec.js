import set from 'lodash/fp/set'
import store from '../../__mocks__/store'
import { InfoPanel, mapStateToProps } from './InfoPanel'

describe('InfoPanel tests', () => {
  it('renders correctly', () => {
    shallowExpect(InfoPanel)(store).toMatchSnapshot()
  })

  it('renders correctly when there is no attached device', () => {
    shallowExpect(InfoPanel)(set('version.firmware', NaN)(store)).toMatchSnapshot()
  })

  it('should utilize mapStateToProps', () => {
    expect(mapStateToProps(store)).toEqual({ version: store.version })
  })
})
