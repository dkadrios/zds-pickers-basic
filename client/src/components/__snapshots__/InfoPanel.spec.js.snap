// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel tests renders correctly 1`] = `
<footer>
  <div>
    © Copyright 
    <a
      href="https://zendrumstudio.com"
    >
      Zendrum Studio
    </a>
    , 
    2019
  </div>
  <div>
    <span>
      Client: 
      v0.3
    </span>
    <span>
      Firmware: 
      v0.1
    </span>
  </div>
</footer>
`;

exports[`InfoPanel tests renders correctly when there is no attached device 1`] = `
<footer>
  <div>
    © Copyright 
    <a
      href="https://zendrumstudio.com"
    >
      Zendrum Studio
    </a>
    , 
    2019
  </div>
  <div>
    <span>
      Client: 
      v0.3
    </span>
    <span>
      Firmware: 
      N/A
    </span>
  </div>
</footer>
`;
